app/views/general_shopping_list/index.html.erb

```diff
@@ -0,0 +1,45 @@
+<%= render 'layouts/navbar' %> 
+
+<div class="max-width flex-center">
+  <div class="row food-heading">
+    <%= render partial: "shared/heading_text" %>
+  </div>
+</div>
+
+
+<div class="max-width flex-center column shopping">
+  <h2>Shopping List</h2>
+  <div class="row max-width detail">
+    <p>
+      Amount of food items to buy: <%= @shop_list.length %>
+    </p>
+    <p>
+      Total value of food neeeded: <%= @total_price %>
+    </p>
+  </div>
+
+  <table class="table">
+    <tr>
+      <th>Food</th>
+      <th>Quantity</th>
+      <th>Price</th>
+    </tr>
+    <% @shop_list.each do |recipe_food| %>
+      <tr>
+        <td><%= recipe_food.name %></td>
+        <td><%= recipe_food.quantity %></td>
+        <td><%= recipe_food.price %></td>
+      </tr>
+    <% end %>
```

Comment:
- [ Required ] Please consider making the `general_shopping` generate a shopping list based on the ingredients that you have and what you may be missing from the recipe. Currently, it displays an empty shopping list yet I have added some ingredients to the recipe. Please see the screenshot below for more details on this.
## Project wireframe
![](https://raw.githubusercontent.com/microverseinc/curriculum-rails/main/recipe-app/images/general_shopping_list.png?token=GHSAT0AAAAAABTYBUDZADCA26RXMOSYTVCIZCLTGVA)


## Your design
![Screenshot from 2023-04-28 11-57-07](https://user-images.githubusercontent.com/16392046/235104457-5b228c98-6141-48c7-ac66-12e6151b318a.png)

Reply:
Done

Reply:
Not resolved ✖️. The shopping list is still empty.

Recipe:
![image](https://user-images.githubusercontent.com/38088818/235138168-8d96eb97-feb7-421e-82a1-626a6e5d9d6d.png)

Shopping list:
![image](https://user-images.githubusercontent.com/38088818/235138247-7c81ed47-f070-4c1c-9ad3-3fa84e5032e7.png)

Reply:
Hi @KDlamini, It's still work on my side.
May be consider checking:

- **Use the same user that own the recipe**: only user recipes are take in count
- **Make sure the recipe requires more quantity than the food**: Only the missing foods should be shown on the shopping list. So If you have more food quantity than in the shopping list it will show an empty list as the user already own all required food (if user.food.quantity > recipe_food.quantity.food.quantity)